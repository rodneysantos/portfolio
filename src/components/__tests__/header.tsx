import { render } from "@testing-library/react";
import React from "react";
import rendeder from "react-test-renderer";

import Header from "../header";

describe("Header", () => {
  it("renders correctly", () => {
    // act
    const tree = rendeder.create(<Header />);

    // assert
    expect(tree).toMatchSnapshot();
  });

  it("displays correct brand name", () => {
    // arrange
    const brandName = "Christopher Santos";

    // act
    const { getByText } = render(<Header />);

    // assert
    expect(getByText(brandName)).toBeDefined();
  });
});
