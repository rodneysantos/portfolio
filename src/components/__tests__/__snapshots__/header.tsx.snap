// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="grid grid-cols-2 p-8"
>
  <div
    className="brand"
  >
    Christopher Santos
  </div>
  <nav
    className="flex"
  >
    <div
      className="gallery flex-1"
    >
      <a
        href="/gallery/black-and-white"
      >
        B&W
      </a>
    </div>
    <div
      className="gallery flex-1"
    >
      <a
        href="/gallery/color"
      >
        Color
      </a>
    </div>
    <div
      className="gallery flex-1"
    >
      <a
        href="/gallery/low-key"
      >
        Low-Key
      </a>
    </div>
    <div
      className="gallery flex-1"
    >
      <a
        href="/gallery/film-simulation"
      >
        Film Simulation
      </a>
    </div>
  </nav>
</header>
`;
